crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -9,6 +9,69 @@ use bevy_ecs::{
 
 use crate::prelude::{Children, Parent, PreviousParent};
 
+fn push_child_unchecked(world: &mut World, parent: Entity, child: Entity) {
+    let mut parent = world.entity_mut(parent);
+    if let Some(mut children) = parent.get_mut::<Children>() {
+        children.0.push(child);
+    } else {
+        parent.insert(Children(smallvec::smallvec![child]));
+    }
+}
+
+fn update_parent(world: &mut World, child: Entity, new_parent: Entity) -> Option<Entity> {
+    let mut child = world.entity_mut(child);
+    if let Some(mut parent) = child.get_mut::<Parent>() {
+        let previous = parent.0;
+        *parent = Parent(new_parent);
+        child.insert(PreviousParent(new_parent));
```

Comment:
```suggestion
        child.insert(PreviousParent(previous));
```
shouldn't that be the previous parent?

Reply:
No. The goal is to hit this line: https://github.com/bevyengine/bevy/blob/86dd6f065d8d355ca6d75ee3ea270b9dad7e8ecd/crates/bevy_hierarchy/src/systems.rs#L34

Reply:
I think it's wrong to set a wrong value to a component because you target a special case for a system that happens after. If you insert the `PreviousParent` component, it should have the correct value at the moment of insertion

Reply:
It does match, that is the correct value. PreviousParent is supposed to match Parent unless someone manually inserts a new Parent.